pyvista/plotting/widgets.py

```diff
@@ -28,15 +28,16 @@
 from .utilities.algorithms import set_algorithm_input
 
 if TYPE_CHECKING:  # pragma: no cover
+    from pyvista.core._typing_core import InteractionEventType
     from pyvista.core._typing_core import VectorLike
 
 
-def _parse_interaction_event(interaction_event):
+def _parse_interaction_event(interaction_event: InteractionEventType):
     """Parse the interaction event.
 
     Parameters
     ----------
-    interaction_event : vtk.vtkCommand.EventIds, str, optional
+    interaction_event : InteractionEventType
```

Comment:
Union-types are auto-expanded by Sphinx. So, in the function signature this will appear as
`Literal['end', 'start', 'always'], _vtk.vtkCommand.EventIds]`, not `InteractionEventType`.

But it won't be auto-expanded in the docstring. So, Sphinx will try to find a ref for `InteractionEventType` which doesn't exist and will cause the build to fail (broken link).

In the docstrings, the type should either be expanded manually, e.g. 
```suggestion
    interaction_event : Literal['end', 'start', 'always'], _vtk.vtkCommand.EventIds
```

Or, you can add the type to the [typing docs](https://dev.pyvista.org/api/core/typing) by doing:
1. Add the type to the "autodoc_type_aliases" dictionary in /doc/source/conf.py
2. Add the type to /doc/core/typing.rst
3. (Optionally) add a short description by defining `InteractionEventType.__doc__` in _`_aliases.py`

This way, Sphinx will be able to find a ref for `InteractionEventType` automatically and auto-link it to the core typing docs.

Reply:
Thanks for the detailed feedback. I added the type in the docs here https://github.com/pyvista/pyvista/pull/6960/commits/f915a4b4be8f6173740bdb37c3caf5182e0af815